openapi.yaml

```diff
@@ -751,8 +751,10 @@ definitions:
         type: string
         description: additional information about using or preparing the ingredient
         example: brown the sausage in a pan before baking
+        default: ''
```

Comment:
Just curious, why can't this be `null`?

Reply:
Because oracledb can't distinguish `null` from `''`, we don't have a way to store an empty string in a way that we can distinguish from a null value, so it's simpler to just only allow one of those values. That way, someone doesn't write code expecting an empty string and then getting `null`, or the other way around.

Reply:
Based on the discussion [here](https://stackoverflow.com/questions/13278773/null-vs-empty-string-in-oracle), I think Oracle will convert the empty strings to null by default, and that's expected. I guess in this case, we shouldn't say the default value of it is an empty string in swagger since it's misleading. You can just simply remove the defualt value and insert null if no value is provided.

Reply:
I don't know if that would be expected from the perspective of an API user. I'm also using a transform function in the serializer so that the API always returns an empty string, even though the database stores it as `null`. I think that's preferable to the case where someone passes in an empty string when they're creating an ingredient record and then gets `null` returned to them in the API even though they were expecting an empty string.

Ian and I discussed this [here](https://github.com/osu-mist/pizza-api/pull/7#pullrequestreview-316390831) with the PR for POST doughs if you want to see his take on it.

Reply:
I'd argue with it. To me, the meaning of empty strings and null are slightly different. Using instruction as an example, an empty string can stand for there is no instruction at all, however, null means either there's no instruction or just not having yet.

Reply:
@Ruefa I guess we need another tiebreaker lol, can you share your thought on it?

Reply:
Plus, I think returning null is very common behavior, so I don't think it's necessary to convert the value to an empty string. We could just return whatever we saw in the database.

Reply:
I guess the last thing I would add is that the openapi spec gives us a way to communicate that an attribute will always store an empty string if there's no value, but no way to communicate that an attribute will either be `null` or a string with length > 1. At least as far as I know.

Reply:
Actually, the last last thing I will add is that in javascript an empty string mostly behaves like null, but null does not always behave like an empty string. So if you're writing frontend code at least, it's worse to get `null` when you're expecting an empty string than to get an empty string when you're expecting `null`.